src/middle/Stan_math_signatures.mli

```diff
@@ -24,7 +24,19 @@ type variadic_signature =
   { return_type: UnsizedType.t
   ; control_args: fun_arg list
   ; required_fn_rt: UnsizedType.t
-  ; required_fn_args: fun_arg list }
+  ; required_fn_args: fun_arg list
+  ; hof_pstream_loc: int
+        (** This specifies where the pstream* argument goes in the call to the variadic function,
+      usually after some required arguments and control args.contents
+
+      For example, for dae(), the arguments [f, yy0, yp0, t0, ts] all precede
+      the msgs argument, making msgs the 6th argument, so this number is 6.
+      *)
+  ; required_fn_pstream_loc: int
+        (** Similar to the above, but specifies where in the {e functor} we expect
+      pstream* to occur. For example, for a dae, this is the 4th argument,
+      so this number is 4 *)
```

Comment:
I suspected there was some relationship but I wasn't sure.

I agree with making them 0-indexed and basing them off those lenghts, but I am still partial toward storing them in this record for easy access/to avoid recomputing them later

Reply:
Each is actually only accessed once or twice, so it's probably better to remove them.